lib/app/ctx.js

```diff
@@ -224,6 +224,10 @@ Box.prototype.unwrap = function (data, content) {
             if (msg.type.substr(0, 3) === 'QLB' && tr.docs.length > 0) {
                 content = tr.docs.shift().contents;
             }
+            if (msg.type === 'DOCUMENT') {
+                data = msg.contents;
```

Comment:
Ah, I see, it was done at line 213, so it would work.

What I don't like here is `continue` statement. Is it possible to:

- make it work without continue statement and without recursion in app (maybe moving this check up and calling transport.decode second time)
- explicitly check message type at line 227. something like `if (msg.type === 'DOCUMENT' && msg.encoding === 'PACKED_XML')` would work.